### PR TITLE
PUBDEV-8868: Expose eval_metric in XGB, use it for early stopping

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

**Comment by michalkurka** [comment]
@valenad1 at this point just for your information - this is still a draft

**Comment by michalkurka** [comment]
@kanech please review

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java [hunk 1]
```diff
@@ -602,10 +610,10 @@ private ModelMetrics makeMetrics(Frame data, Frame originalData, boolean isTrain
    * @param _train training data in the form of matrix
    * @param _valid validation data (optional, can be null)
    */
-  final void doScoring(Frame _train, Frame _trainOrig, Frame _valid, Frame _validOrig) {
+  final void doScoring(Frame _train, Frame _trainOrig, CustomMetric trainCustomMetric, Frame _valid, Frame _validOrig) {
```
Comment (kanech) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java:613]: Odd parameter naming convention. I don't know why but all existing parameter names start with underscore. Adding a new parameter, without underscore, leads to even more weird situation.

**Review by kanech — APPROVED** [review]
LGTM

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java (hunk 1 above)
Comment (michalkurka) [reply h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java:613]: good point, thanks for catching that - I will fix it

**Pull request merged**